source/extensions/priv/server/ntds_jet.c

```diff
@@ -0,0 +1,312 @@
+#include "precomp.h"
+
+JET_ERR engine_shutdown(jetState *ntdsState){
+	JET_ERR shutdownStatus;
+	shutdownStatus = JetCloseDatabase(ntdsState->jetSession, ntdsState->jetDatabase, (JET_GRBIT)NULL);
+	if (shutdownStatus != JET_errSuccess){
+		return shutdownStatus;
+	}
+	shutdownStatus = JetDetachDatabase(ntdsState->jetSession, ntdsState->ntdsPath);
+	if (shutdownStatus != JET_errSuccess){
+		return shutdownStatus;
+	}
+	shutdownStatus = JetEndSession(ntdsState->jetSession, (JET_GRBIT)NULL);
+	if (shutdownStatus != JET_errSuccess){
+		return shutdownStatus;
+	}
+	shutdownStatus = JetTerm(ntdsState->jetEngine);
+	free(ntdsState);
+	return shutdownStatus;
+}
+
+JET_ERR engine_startup(jetState *ntdsState){

```

Comment:
Same here with some function docs, so that people know that the part of the initialisation is their responsibility?